wdn/templates_4.0/scripts/search.js

```diff
@@ -151,15 +156,49 @@ define(['jquery', 'wdn', 'require', 'modernizr', 'navigation'], function($, WDN,
 					if (postReady && $unlSearch[0].contentWindow.postMessage) {
 						e.preventDefault();
 						$unlSearch[0].contentWindow.postMessage(domQ.val(), searchOrigin);
+						if ('auto' != source) {
+							//a11y: send focus to the results if manually submitted
+							$unlSearch.focus();
+						}
 						$progress.hide();
 					}
 				});
 
+				var closeSearch = function() {
+					var $wdnSearch = domSearchForm.parent();
+					$wdnSearch.removeClass('active');
+				};
+
+				//Close search on escape while the iframe has focus
+				$(window).on('message', function(e) {
+					var originalEvent = e.originalEvent;
+					
+					if ('wdn.search.close' != originalEvent.data) {
+						//Make sure this is our event
+						return;
+					}
+					
+					if (searchOrigin != originalEvent.origin) {
+						//Verify the origin
+						return;
+					}
+					
+					closeSearch();
+				});
+
+				//Close search on escape
+				$(document).on('keydown', function(e) {
+					if (!e.keyCode || e.keyCode === 27) {
```

Comment:
Good question.  I saw it referenced in a few examples and thought I'd try to find a case where it was helpful.  I couldn't find a good case for it and forgot to take it out.